### ISSUE TITLE
[feature-request] Vim plugin should provide option to search through all files in current project dir

### ISSUE THREAD

**Issue by natemara** [junegunn/fzf#369]
As in #368, this is a clone of a feature existing in CtrlP. Imagine I'm working on a project with this structure:

├── .git
├── module1
│   ├── bang
├── module2
│   ├── boom
└── module3
    └── twang

If I am editing `module3/twang`, and I use FZF, I cannot see any files outside of `module3/`. CtrlP will traverse up the directory tree until it finds a `.git`, `.svn`, `.hv`, etc. folder. This means that any project that is under version control can easily be traversed using only this one navigation tool. Would it be possible to add something similar to FZF? If not in the standard options, perhaps with some additional configuration by the users?

**Comment by justinmk** [comment]
Are you using `autochdir`? If so I suggest not doing that, it's a legacy option that should not be used.

**Comment by junegunn** [comment]
Something like this?

```vim
function! s:find_root()
  for vcs in ['.git', '.svn', '.hg']
    let dir = finddir(vcs.'/..', ';')
    if !empty(dir)
      execute 'FZF' dir
      return
    endif
  endfor
  FZF
endfunction

command! FZFR call s:find_root()
```

- `FZF` -> `Files` if you have fzf.vim project
- Using VCS-specific commands to find the root should give more accurate results
    - e.g. `git rev-parse --show-toplevel`
- You can consider using the low-level `fzf#run()` directly if you need more control

**Comment by natemara** [comment]
Thanks guys.

Closed #369
